webapps/frontend/app/viewmodels/wizards/createfromtemplatewizard/gather.js

```diff
@@ -155,6 +176,16 @@ define([
                 self.data.vm().load();
                 self.data.selectedPMachines([]);
             }
+            generic.xhrAbort(self.loadVMachinesHandle);
+            self.loadVMachinesHandle = api.get('vmachines/')
```

Comment:
You could add a ```contents = 'name'``` here so you only have to do one call instead of possibly hundreds. That would take that long that the check doesn't make much sense (you most likely already completed the wizard before the calls are completed)